tests/Integration/Webservice/AddressesTest.php

```diff
@@ -0,0 +1,315 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Webservice;
+
+use Address;
+use Symfony\Component\DomCrawler\Crawler;
+use WebserviceKey;
+
+class AddressesTest extends AbstractWebserviceTest
+{
+    private const INITIAL_COUNT = 5;
+
+    public function testReturnEmptyWSKey(): void
+    {
+        $output = $this->requestWebserviceXML('', 'GET', 'addresses');
+
+        self::assertEquals(1, $output->filter('prestashop > errors > error')->count());
+        self::assertEquals(17, $output->filter('prestashop > errors > error')->first()->filter('code')->text());
+        self::assertEquals(
+            'Authentication key is empty',
+            $output->filter('prestashop > errors > error')->first()->filter('message')->text()
+        );
+    }
+
+    public function testReturnBadKey(): void
+    {
+        $output = $this->requestWebserviceXML('ABCDE', 'GET', 'addresses');
+
+        self::assertEquals(1, $output->filter('prestashop > errors > error')->count());
+        self::assertEquals(18, $output->filter('prestashop > errors > error')->first()->filter('code')->text());
+        self::assertEquals(
+            'Invalid authentication key format',
+            $output->filter('prestashop > errors > error')->first()->filter('message')->text()
+        );
+    }
+
+    public function testReturnDisabledKey(): void
+    {
+        $this->wsKey->active = false;
+        $this->wsKey->save();
+
+        $output = $this->requestWebserviceXML($this->wsKey->key, 'GET', 'addresses');
+
+        self::assertEquals(2, $output->filter('prestashop > errors > error')->count());
+        self::assertEquals(20, $output->filter('prestashop > errors > error')->first()->filter('code')->text());
+        self::assertEquals(
+            'Authentification key is not active',
+            $output->filter('prestashop > errors > error')->first()->filter('message')->text()
+        );
+
+        self::assertEquals(21, $output->filter('prestashop > errors > error')->last()->filter('code')->text());
```

Comment:
Some repetitions to crawl to the data we want to test. Can we have helpers for that
like 'getErrors', 'getCode' ...